src/iceflow/data/models.py

```diff
@@ -44,3 +46,20 @@ class ATM1BSchema(CommonDataColumnsSchema):
 
 IceflowDataFrame = DataFrame_co[CommonDataColumnsSchema]
 ATM1BDataFrame = DataFrame_co[ATM1BSchema]
+
+DatasetShortName = Literal["ILATM1B"]
+
+
+class Dataset(pydantic.BaseModel):
+    short_name: DatasetShortName
+    version: str
+
+
+class ATM1BDataset(Dataset):
+    short_name: DatasetShortName = "ILATM1B"
+
+
+class DatasetSearchParameters(pydantic.BaseModel):
+    dataset: Dataset
+    bounding_box: Sequence[float]
+    temporal: tuple[str, str]
```

Comment:
What about `tuple[datetime, datetime]`? The type can be coerced from string, I think by default?

Reply:
Good suggestion!

Reply:
e9fd548 * origin/pydantic-for-datasets Type temporal range as tuple[date, date]

I typed this as `tuple[dt.datetime | dt.date, dt.datetime | dt.date]`, as this is passed straight to `earthaccess` and it seems to support both. I like being able to just specify a date range without times, but users may want to specify time ranges too. 

Do you have thoughts on if I should just make this accept datetimes only vs dates & datetimes? I